### ISSUE TITLE
Transparent black in SVG export appears as black

### ISSUE THREAD

**Issue by hauswirth** [LuCEresearchlab/pytamaro#24]
Colors such as `rgb_color(0, 0, 0, 0.5)` are turned into fully opaque black when exported to SVG.

Colors like `rgb_color(255, 255, 255, 0.5)` are properly turned into partially transparent colors.

Example:

```python
from pytamaro import rectangle, red, rgb_color, above, beside, show_graphic, save_graphic

b000 = rectangle(100, 50, rgb_color(0, 0, 0, 0.0))
b050 = rectangle(100, 50, rgb_color(0, 0, 0, 0.5))
b100 = rectangle(100, 50, rgb_color(0, 0, 0, 1))

w000 = rectangle(100, 50, rgb_color(255, 255, 255, 0.0))
w050 = rectangle(100, 50, rgb_color(255, 255, 255, 0.5))
w100 = rectangle(100, 50, rgb_color(255, 255, 255, 1))

top = beside(b000, beside(b050, b100))
bot = beside(w000, beside(w050, w100))

demo = overlay(
    above(top, bot), 
    rectangle(400, 200, red)
)
show_graphic(demo)
save_graphic("demo.svg", demo)
```

Bitmap:
![image](https://github.com/LuCEresearchlab/pytamaro/assets/583696/7868418b-6c55-4ccb-93b2-e10409d8b53b)

SVG:
![demo](https://github.com/LuCEresearchlab/pytamaro/assets/583696/0c6ef61d-fc89-497f-9f55-79ad3cde4f2b)

**Comment by hauswirth** [comment]
Here is the source of the exported SVG:

```svg
<?xml version="1.0" encoding="utf-8" ?>
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="400" height="200" shape-rendering="crispEdges">
	<path fill="red" d="M0 0L400 0L400 200L0 200L0 0Z"/>
	<path fill="white" transform="translate(250 100)" d="M0 0L100 0L100 50L0 50L0 0Z"/>
	<path fill="white" fill-opacity="0.50196081" transform="translate(150 100)" d="M0 0L100 0L100 50L0 50L0 0Z"/>
	<path fill="white" fill-opacity="0" transform="translate(50 100)" d="M0 0L100 0L100 50L0 50L0 0Z"/>
	<path transform="translate(250 50)" d="M0 0L100 0L100 50L0 50L0 0Z"/>
	<path transform="translate(150 50)" d="M0 0L100 0L100 50L0 50L0 0Z"/>
	<path transform="translate(50 50)" d="M0 0L100 0L100 50L0 50L0 0Z"/>
</svg>
```

**Comment by lucach** [comment]
This is an upstream bug in skia (!!!). SVGCanvas is somewhat experimental. It annoyed me enough that I tracked down the root cause.

> [These lines in `SkSVGDevice` are responsible for the bug](https://github.com/google/skia/blob/e5a6f9e79b0aefcfc461748e8a0f3c10f7722e84/src/svg/SkSVGDevice.cpp#L369-L376). The `fill` attribute is not added when it coincides with the default fill (black). However, in that case, the `fill-opacity` attribute is also not added, even when it differs from opaque.
> 
> This is actually a regression: [this commit](https://github.com/google/skia/commit/05bab9a29d621cb155c4fba80d7e8bb50927bdd7) introduced the outermost check, wrongly nesting the check for the opacity inside it.


I've [filed an issue on their bug tracker](https://issues.skia.org/issues/309068101).

**Comment by bvlj** [comment]
Pushed a change to the Skia gerrit: https://skia-review.googlesource.com/c/skia/+/780277

**Comment by bvlj** [comment]
Our fix got accepted and is now merged upstream: https://github.com/google/skia/commit/4c964f8c4738a0e38c0fe15bb3dbc5c2849236c2

Now we need to wait until it makes to the skia python library

**Comment by hauswirth** [comment]
Excellent! Congrats!

**Comment by lucach** [comment]
🎉  Finally fixed in d390ea1537a9d32eea8b6f01b2bfc1c521e8b694 by upgrading to the latest `skia-python` (alongside other improvements to the generation of SVGs).
